client/go/cmd/config.go

```diff
@@ -179,19 +183,41 @@ func (c *Config) X509KeyPair(app vespa.ApplicationID) (KeyPair, error) {
 }
 
 func (c *Config) APIKeyPath(tenantName string) string {
-	if override, ok := os.LookupEnv("VESPA_CLI_API_KEY_FILE"); ok {
+	if override, err := c.Get(apiKeyFileFlag); err == nil && override != "" {
 		return override
 	}
 	return filepath.Join(c.Home, tenantName+".api-key.pem")
 }
 
 func (c *Config) ReadAPIKey(tenantName string) ([]byte, error) {
-	if override, ok := os.LookupEnv("VESPA_CLI_API_KEY"); ok {
+	if override, err := c.Get(apiKeyFlag); err == nil && override != "" {
 		return []byte(override), nil
 	}
 	return ioutil.ReadFile(c.APIKeyPath(tenantName))
 }
 
+// UseAPIKey checks if api key should be used be checking if api-key or api-key-file has been set.
+func (c *Config) UseAPIKey(tenantName string) bool {
+	if _, err := c.Get(apiKeyFlag); err == nil {
+		return true
+	}
+	if _, err := c.Get(apiKeyFileFlag); err == nil {
+		return true
+	}
+
+	// If no Auth0 token is created, fall back to tenant api key, but warn that this functionality is deprecated
+	a, err := auth0.GetAuth0(c.AuthConfigPath(), getSystemName(), getApiURL())
+	if err != nil || !a.HasSystem() {
```

Comment:
Why do we need all these checks? Shouldn't we simply use the API key if it exists and print a warning that the user should setup Auth0 with `vespa auth login`?

Reply:
This was never meant as a permanent solution, more of a way to support the old behaviour for some releases to nudge people to the device flow. It will give people some time to migrate over, but I am happy to rip it all out if we think this functionality is not part of any automated systems somewhere.